src/Universum/Exception.hs

```diff
@@ -19,6 +24,25 @@ import Control.Applicative (Applicative (pure))
 import Control.Monad.Except (MonadError, throwError)
 import Data.Maybe (Maybe, maybe)
 
+#if ( __GLASGOW_HASKELL__ >= 800 )
+import GHC.Show (Show)
+import GHC.Stack (CallStack, HasCallStack)
+
+import qualified Control.Exception.Safe as Safe (impureThrow, toException)
+
+-- | Type that represents exceptions used in cases when a particular codepath
+-- is not meant to be ever executed, but happens to be executed anyway.
+data Bug = Bug SomeException CallStack
+    deriving (Show)
+
+instance Exception Bug
+
+-- | Generate a pure value which, when forced, will synchronously
+-- throw the exception wrapped into 'Bug' data type.
+bug :: (HasCallStack, Exception e) => e -> a
+bug e = Safe.impureThrow (Bug (Safe.toException e) ?callStack)
```

Comment:
I found this note in `base` docs:

![image](https://user-images.githubusercontent.com/4061728/34279333-5ae3d564-e6c1-11e7-8dfa-f57d9fdf7be0.png)

Instead of `?callStack`, we should use the `callStack` function from `GHC.Stack`.

Reply:
Then the `ImplicitParams` extension can be removed.